### PR TITLE
When there are extra parentheses, display the exn

### PR DESCRIPTION
**Pull request by emillon**
Dune tries to be helpful when there are extra parentheses, but sometimes it is a bit too eager.
This wraps the exception so that the original one is displayed.

I'm not completely satisfied with the output, so let me know what you think.

### CODE REVIEW

File: test/blackbox-tests/test-cases/too-many-parens/run.t [hunk 1]
```diff
@@ -8,25 +8,39 @@ are readable.
   3:   (public_name hello)
   4:   (libraries (lib))
   5: ))
-  These parentheses are no longer necessary with dune, please remove them.
+  Error: Atom expected
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
```
Comment (diml) [inline test/blackbox-tests/test-cases/too-many-parens/run.t]: The ` try:` feels out of place here

File: test/blackbox-tests/test-cases/too-many-parens/run.t [hunk 2]
```diff
@@ -8,25 +8,39 @@ are readable.
...
+  [1]
+
+When the inner syntax is wrong, do not warn about the parens:
+
+  $ dune build --root e
+  File "dune", line 3, characters 7-15:
+   (deps (glob *)) ; this form doesn't exist
+         ^^^^^^^^
+  Error: Unknown constructor glob
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
```
Comment (diml) [inline test/blackbox-tests/test-cases/too-many-parens/run.t]: In this particular case, the hint doesn't point the user in the right direction. It seems hard to fix in general, maybe we should just make the message a bit more generic:

```
Hint: dune files require less parentheses than jbuild files.
If you just converted this file from a jbuild file, try removing these parentheses.
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by diml** [comment]
BTW, I noticed that some error messages prefix the printed source code with line numbers, while some don't. Is that expected?

File: test/blackbox-tests/test-cases/too-many-parens/run.t (hunk 1 above)
Comment (emillon) [reply test/blackbox-tests/test-cases/too-many-parens/run.t]: Yes, it's hardcoded in `Hint`. I can move this.

File: test/blackbox-tests/test-cases/too-many-parens/run.t (hunk 2 above)
Comment (emillon) [reply test/blackbox-tests/test-cases/too-many-parens/run.t]: So, we'd display the "unknown constructor" error and this hint? sounds good to me.

**Comment by emillon** [comment]
The line numbers are only printed when there are several lines, but it's true that we should always print them for consistency.

File: test/blackbox-tests/test-cases/too-many-parens/run.t (hunk 2 above)
Comment (diml) [reply test/blackbox-tests/test-cases/too-many-parens/run.t]: yep. This is basically a best effort to help the user fix the problem

**Comment by emillon** [comment]
Done! Let me know what you think.

**Review by ghost — APPROVED** [review]

**Comment by diml** [comment]
Looks good!

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**